api/src/wfl/module/covid.clj

```diff
@@ -263,6 +248,49 @@
                     (->> (jdbc/insert! tx terra-executor-table)))]
     [terra-executor-type (-> items first :id)]))
 
+(defn import-snapshot!
+  "Return snapshot reference for SNAPSHOT_ID imported to WORKSPACE."
+  [{:keys [workspace]   :as _executor}
+   {:keys [snapshot_id] :as _snapshot}]
+  (rawls/create-snapshot-reference workspace snapshot_id))
+
+(defn from-source
+  "Coerce SOURCE-ITEM to form understood by EXECUTOR via FROMSOURCE."
+  [{:keys [fromSource] :as executor}
+   source-item]
+  (cond (= "importSnapshot" fromSource) (import-snapshot! executor source-item)
+        :else (throw (ex-info "Unknown fromSource" {:executor executor}))))
+
+(defn ^:private update-terra-executor
```

Comment:
Right now, this method may trigger a submission from a coerced member of the `source` queue.  We also want it to check the status for any active workflows in the `executor` queue and update them if necessary.  This is similar to how @rexwangcc is expanding `update-tdr-source`.

I'd like to save this for a future PR.